### PR TITLE
Filter /institutions by user auth

### LINKED ISSUE #331: API - List of institutions for a user
`/institutions/period/<period>` (period is being used in this case to represent a time period, year or year and quarter)

`GET` - Complete list of institutions a user can/could file for a given year.

[Example](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/user1-institutions.json)

### LINKED ISSUE #412: Filter list of institutions by user
See #331 
The list of institutions returned needs to be filtered according to the user that is logged in. The approach that has been discussed at this point is that the list of institutions to be returned will be part of a request header

### PR DESCRIPTION
**Pull request by schbetsy**
Closes #412 (I believe. See comment in issue)
Closes #331

### CODE REVIEW

**Comment by schbetsy** [comment]
CodeCov fails because the failure case of this endpoint (and all of our other endpoints), where it returns a 500 error, isn't covered with tests.

**Comment by nickgrippin** [comment]
LGTM

File: persistence/src/test/scala/hmda/persistence/institutions/InstitutionPersistenceSpec.scala [hunk 1]
```diff
@@ -34,6 +34,24 @@ class InstitutionPersistenceSpec extends ActorSpec {
...
       probe.expectMsg(modified)
     }
+
+    "return a set of institutions matching a list of ids" in {
+      // Setup: Persist demo institutions
+      val i1 = Institution(71, "CFPBank", Set(ExternalId("99-1234567", FederalTaxId)), CFPB, Bank, hasParent = true, Active)
+      val i2 = Institution(72, "FRBank", Set(ExternalId("654321", RssdId)), CFPB, Bank, hasParent = true, Active)
+      val i3 = Institution(73, "MLBank", Set(ExternalId("externalTest0", FdicCertNo)), FDIC, Bank, hasParent = true, Active)
+      for (institution <- List(i1, i2, i3)) {
+        probe.send(institutionsActor, CreateInstitution(institution))
```
Comment (kgudel) [inline persistence/src/test/scala/hmda/persistence/institutions/InstitutionPersistenceSpec.scala]: Is there a reason for using these specific institutions over the test institutions?

Comment (schbetsy) [inline persistence/src/test/scala/hmda/persistence/institutions/InstitutionPersistenceSpec.scala]: My reason (tbd whether it's a good reason :) ) was that `DemoData.testInstitutions` is a `Set`, and I don't know of a good way to get a reference to an individual member of a set. That is, I wanted to call `GetInstitutionsById` with just 2 of the 4 institutions, but I couldn't find a good way to set up my expectation (line 49) if I did that.

Comment (kgudel) [inline persistence/src/test/scala/hmda/persistence/institutions/InstitutionPersistenceSpec.scala]: You could always convert it to a list with `DemoData.testInstitutions.toList` or another type that is easier to work with :)!

Comment (schbetsy) [inline persistence/src/test/scala/hmda/persistence/institutions/InstitutionPersistenceSpec.scala]: OH YEAH. So I just tried your suggestion (good suggestion, by the way!), and I remembered the real reason why I had to create my own data with new ids. In the collection of persisted Institutions, there are already multiple records with id 0, 1, 2, and 3. So when I use `GetInstitutionsById`, I get multiple responses for each `id`. The fact that we allow duplicate IDs is a separate issue, so I'll create a GH issue for it.

**Pull request merged**